### PR TITLE
Add Expanding and EWM.mean

### PR DESCRIPTION
**Pull request by mroeschke**
xref #220

Added `expanding` and `ewm` ops. For `ewm`, I used pandas' ewm implementation.

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
@mroeschke if u can add some tests esp to assert the NotImplemented

File: streamz/dataframe/core.py [hunk 1]
```diff
@@ -622,6 +640,122 @@ def groupby(self, other):
...
+            if halflife <= 0:
+                raise ValueError("halflife must satisfy: halflife > 0")
+            decay = 1 - np.exp(np.log(0.5) / halflife)
+            com = 1 / decay - 1
+        elif alpha is not None:
+            if alpha <= 0 or alpha > 1:
+                raise ValueError("alpha must satisfy: 0 < alpha <= 1")
+            com = (1 - alpha) / alpha
+        else:
+            raise ValueError("Must pass one of com, span, halflife, or alpha")
```
Comment (jreback) [inline streamz/dataframe/core.py:718]: are these all hit in tests?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by ramicaza** [comment]
I started working on implementing EWMA today then halfway through I realized "hey I should probably check if there were any PRs in the last month since I last checked".  Then I found this haha. 

Anyways, I noticed there was no mention of "adjust" in your changes. My implementation currently computes a running adjusted EWM which is equivalent to the default pandas ewm: `pd.DataFrame.ewm(adjust=True)`.  I was wondering if your implementation is the adjusted or non-adjusted ewma.  I think it's pretty important to offer both of these in the streamz ewm.

File: streamz/dataframe/core.py (hunk 1 above)
Comment (mroeschke) [reply streamz/dataframe/core.py:718]: Added tests for these.

**Comment by mroeschke** [comment]
Agreed, and it would be great if you could follow up with those features @ramicaza! I don't think I'll be able to tackle that if/when this PR is merged in.

**Comment by mroeschke** [comment]
@martindurant ready for a look when you get the chance (the failing test seems unrelated `test_kafka_batch_checkpointing_async_nodes_1`)

**Comment by martindurant** [comment]
Agreed, same as in https://github.com/python-streamz/streamz/pull/411

**Pull request merged**